pkg/daemon/daemon.go

```diff
@@ -1194,21 +1199,19 @@ func (dn *Daemon) prepUpdateFromCluster() (*mcfgv1.MachineConfig, *mcfgv1.Machin
 	if err != nil {
 		return nil, nil, err
 	}
-	currentConfig, err := dn.mcLister.Get(currentConfigName)
+	currentConfig, err := dn.getCurrentConfig(currentConfigName)
 	if err != nil {
 		return nil, nil, err
 	}
 	state, err := getNodeAnnotation(dn.node, constants.MachineConfigDaemonStateAnnotationKey)
 	if err != nil {
 		return nil, nil, err
 	}
-
 	currentOnDisk, err := dn.getCurrentConfigOnDisk()
 	if err != nil && !os.IsNotExist(err) {
 		return nil, nil, err
 	}
-
-	if currentOnDisk != nil && currentOnDisk.GetName() != currentConfig.GetName() {
+	if currentOnDisk != nil && currentOnDisk.GetName() != currentConfigName {
```

Comment:
Nit: In the future it may make sense to functionize this check as it happens multiple times.